content/docs/forwarding-refs.md

```diff
@@ -1,76 +1,78 @@
 ---
 id: forwarding-refs
-title: Forwarding Refs
+title: Ref'leri yönlendirme
 permalink: docs/forwarding-refs.html
 ---
 
-Ref forwarding is a technique for automatically passing a [ref](/docs/refs-and-the-dom.html) through a component to one of its children. This is typically not necessary for most components in the application. However, it can be useful for some kinds of components, especially in reusable component libraries. The most common scenarios are described below.
+Ref yönlendirme, bir [ref](/docs/refs-and-the-dom.html)'i üst bileşenlerden alt bileşenlerin birine otomatik olarak aktarma tekniğidir. Bu genellikle uygulamadaki çoğu bileşen için gerekli değildir. Ama, bazı bileşen türleri için faydalı olabilir, özellikle yeniden kullanılabilir bileşen kütüphaneleri. En yaygın senaryolar aşağıda açıklanmaktadır.
 
-## Forwarding refs to DOM components {#forwarding-refs-to-dom-components}
+## Ref'leri DOM bileşenlerine aktarmak {#forwarding-refs-to-dom-components}
 
-Consider a `FancyButton` component that renders the native `button` DOM element:
+Yerel (native) `button` öğesisini oluşturan `FancyButton` bileşeni düşünün:
 `embed:forwarding-refs/fancy-button-simple.js`
 
-React components hide their implementation details, including their rendered output. Other components using `FancyButton` **usually will not need to** [obtain a ref](/docs/refs-and-the-dom.html) to the inner `button` DOM element. This is good because it prevents components from relying on each other's DOM structure too much.
+React bileşenleri, uygulama ayrıntılarını gizler. `FancyButton` bileşenini kullanan diğer bileşenler, genellikle, inner `button` için [ref oluşturması](/docs/refs-and-the-dom.html) **gerekmez**. Bu iyi bir şeydir, çünkü bileşenlerin birbirilerinin DOM yapısına fazla bağımlı olmasını önler.
```

Comment:
*There is a lot of missing translation here. Here is a more appropriate translation for start:*
React bileşenleri, render edilen çıktıları da dahil olacak bir şekilde uygulama ayrıntılarını gizler. `FancyButton` bileşenini kullanan diğer bileşenler, alt `button` DOM elemanı için **genellikle gerekmeyen** [ref elde ederler](/docs/refs-and-the-dom.html).